packages/datadog-instrumentations/src/kafkajs.js

```diff
@@ -0,0 +1,112 @@
+'use strict'
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+const producerStartCh = channel('apm:kafkajs:produce:start')
+const producerMessageCh = channel(`apm:kafkajs:produce:message`)
+const consumerStartCh = channel('apm:kafkajs:consume:start')
+
+const consumerFinishCh = channel('apm:kafkajs:consume:finish')
+const consumerErrorCh = channel('apm:kafkajs:consume:error')
+
+const producerFinishCh = channel('apm:kafkajs:produce:finish')
+const producerErrorCh = channel('apm:kafkajs:produce:error')
+
+addHook({ name: 'kafkajs', versions: ['>=1.4'] }, (obj) => {
+  const Kafka = obj.Kafka
+  shimmer.wrap(Kafka.prototype, 'producer', createProducer => function () {
+    if (!producerStartCh.hasSubscribers) {
+      return createProducer.apply(this, arguments)
+    }
+    const producer = createProducer.apply(this, arguments)
+    const send = producer.send
+
+    producer.send = wrapProducer(function (...args) {
```

Comment:
Why is `wrapProducer` needed? It looks like a lot more code compared to the original instrumentation.